playground/src/main/java/mx/yellowme/youst/playground/ui/chart/utils/ChartBuilder.kt

```diff
@@ -0,0 +1,60 @@
+package mx.yellowme.youst.playground.ui.chart.utils
+
+import androidx.fragment.app.FragmentActivity
+import com.github.mikephil.charting.charts.BarChart
+import com.github.mikephil.charting.charts.BarLineChartBase
+import com.github.mikephil.charting.charts.BubbleChart
+import com.github.mikephil.charting.charts.LineChart
+import mx.yellowme.youst.core.utils.loadJsonObject
+import mx.yellowme.youst.playground.domain.ChartSetting
+import mx.yellowme.youst.playground.domain.ChartType
+
+/**
+ * @author adrianleyvasanchez
+ * @since 03,October,2019
+ */
+object ChartBuilder {
+
+    fun buildChart(activity: FragmentActivity): BarLineChartBase<*>? {
+        with(activity) {
+            parseSettingsJson(this).run {
+                instanceTypeChart(this, activity).also {
+                    return setConfiguration(this, it, activity)
+                }
+            }
+        }
+    }
+
+    fun buildChartByType(activity: FragmentActivity, type: ChartType): BarLineChartBase<*>? {
+        with(activity) {
```

Comment:
¿Qué te parece si quitamos el `with(activity)`? Solo estamos accediendo una vez y estamos utilizando el this para pasar el valor, podríamos pasar el activity que ya tenemos.

Y utilizando el run podemos hacer lo siguiente. 

```
    fun buildChartByType(activity: FragmentActivity, type: ChartType): BarLineChartBase<*>? =
        parseSettingsJson(activity).run {
            this?.type = type.toString()
            instanceTypeChart(this, activity).also {
                setConfiguration(this, it, activity)
            }
        }
```